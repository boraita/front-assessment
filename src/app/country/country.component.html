<div class="main">
    <app-country-list class="country-list" (countrySelected)="country = $event"></app-country-list>
    <app-show-city [country]="country"></app-show-city>
</div>
  