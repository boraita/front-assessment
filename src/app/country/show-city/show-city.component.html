<div class="container">
  <div class="capital-container">
    <div class="select-country" *ngIf="country == undefined; else capital">
      Select a country
    </div>
    <ng-template #capital>
      <div class="capital-selected">
        <div class="country-name">
          <span> Country </span> <span>{{ country.country }}</span>
        </div>
        <div class="city-name">
          <span> Capital </span> <span> {{ country?.city || "Unknown" }} </span>
        </div>
      </div>
    </ng-template>
  </div>
</div>
