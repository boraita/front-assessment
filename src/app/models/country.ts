export type Country = {
    country: string;
    city: string;
}