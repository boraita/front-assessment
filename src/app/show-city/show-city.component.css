.select-country {
  display: flex;
  justify-content: center;
  border-radius: 5px;
  padding: 10px;
  box-shadow: 0px 0px 5px #000;
  font-weight: bold;
  width: 100%;
}
.capital-selected {
  display: flex;
  flex-direction: column;
  font-weight: bold;
  border-radius: 5px;
  padding: 10px;
  box-shadow: 0px 0px 5px #000;
  font-weight: bold;
  width: 100%;
}
.country-name {
  display: flex;
  justify-content: space-around;
  margin-bottom: 15px;
  font-size: 2em;
  border: 2px solid;
  border-color: gray;
  border-radius: 5px;
}
.city-name {
  display: flex;
  justify-content: space-around;
  margin-bottom: 15px;
  font-size: 2em;
  border: 2px solid;
  border-color: grey;
  border-radius: 5px;
}
.title {
  display: flex;
  justify-content: center;
  align-items: stretch;
  padding: 20px;
  font-weight: bold;
  font-size: large;
  background-color: beige;
}
.capital-container {
  min-width: 20vw;
  left: 50%;
  top: 50%;
  width: auto;
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  padding: 1.6rem 3rem;
  border: 3px solid black;
  border-radius: 5px;
  background: white;
  box-shadow: 8px 8px 0 rgba(0, 0, 0, 0.2);
}

.container {
  display: flex;
  flex-direction: column;
}
@media only screen and (max-width: 1028px) {
  .capital-container {
    min-width: 60vw;
    flex-direction: column;
    align-content: stretch;
  }
  .capital-selected {
    align-content: stretch;
    flex-direction: row;
  }
  .capital-selected > * {
    flex: 0 0 48%;
  }
  .country-name {
    flex-direction: column;
    align-items: center;
  }
  .city-name {
    flex-direction: column;
    align-items: center;
  }
}
