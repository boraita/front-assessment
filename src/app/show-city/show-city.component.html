<div class="select-city" *ngIf="country == undefined; else capital">
  Select a city
</div>
<ng-template #capital>
  <div class="capital-selected">Country selected: {{country.country}} - Capital: {{ country?.city || "Unknown" }}</div>
</ng-template>
