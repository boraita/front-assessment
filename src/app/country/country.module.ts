import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CountryListComponent } from './country-list/country-list.component';
import { ShowCityComponent } from './show-city/show-city.component';
import { HttpClientModule } from '@angular/common/http';
import { CountryService } from './services/country.service';
import { CountryComponent } from './country.component';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [CountryListComponent, ShowCityComponent, CountryComponent],
  imports: [
    CommonModule,
    HttpClientModule,
    RouterModule.forChild([
      {
        path: '',
        component: CountryComponent,
      },
    ]),
  ],
  providers: [CountryService],
})
export class CountryModule {}
