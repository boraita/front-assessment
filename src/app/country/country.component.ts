import { Component } from '@angular/core';
import { Country } from '../models/country';

@Component({
  selector: 'app-country',
  templateUrl: './country.component.html',
  styleUrls: ['./country.component.css'],
})
export class CountryComponent {
  country!: Country;
}
