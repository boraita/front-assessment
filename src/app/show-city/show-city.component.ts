import { Component, Input } from '@angular/core';
import { Country } from '../models/country';

@Component({
  selector: 'app-show-city',
  templateUrl: './show-city.component.html',
  styleUrls: ['./show-city.component.css'],
})
export class ShowCityComponent {
  @Input() country!: Country;
}
