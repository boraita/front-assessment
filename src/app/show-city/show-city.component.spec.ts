import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { ShowCityComponent } from './show-city.component';
import { Country } from '../models/country';

describe('ShowCityComponent', () => {
  let component: ShowCityComponent;
  let fixture: ComponentFixture<ShowCityComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ShowCityComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ShowCityComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('country should be undefined', () => {
    expect(component.country).toBeUndefined();
    const divSelectCity = fixture.debugElement.query(By.css('.select-city'));
    expect(divSelectCity.nativeElement.textContent.trim()).toBe('Select a city');
  });

  it('should be shown capital Unknown', () => {
    component.country = { city: null, country: 'Spain' } as Country;
    fixture.detectChanges();
    const divSelectedCountry = fixture.debugElement.query(By.css('.capital-selected'));
    expect(divSelectedCountry.nativeElement.textContent.trim()).toBe('Country selected: Spain - Capital: Unknown');
  });

  it('should be shown capital from country selected', () => {
    component.country = { city: 'Madrid', country: 'Spain' } as Country;
    fixture.detectChanges();
    const divSelectedCountry = fixture.debugElement.query(By.css('.capital-selected'));
    expect(divSelectedCountry.nativeElement.textContent.trim()).toBe('Country selected: Spain - Capital: Madrid');
  });

});
