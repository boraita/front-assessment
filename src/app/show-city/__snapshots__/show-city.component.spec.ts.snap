// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowCityComponent should display Select a country 1`] = `
<app-show-city
  __ngContext__={[Function LRootView]}
>
  <div
    class="container"
  >
    <div
      class="capital-container"
    >
      <div
        class="select-country"
      >
         Select a country 
      </div>
    </div>
  </div>
</app-show-city>
`;

exports[`ShowCityComponent should display a capital from input country value 1`] = `
<app-show-city
  __ngContext__={[Function LRootView]}
  country={[Function Object]}
>
  <div
    class="container"
  >
    <div
      class="capital-container"
    >
      <div
        class="capital-selected"
      >
        <div
          class="country-name"
        >
          <span>
             Country 
          </span>
          <span>
            Spain
          </span>
        </div>
        <div
          class="city-name"
        >
          <span>
             Capital 
          </span>
          <span>
             Madrid 
          </span>
        </div>
      </div>
    </div>
  </div>
</app-show-city>
`;
